### PR TITLE
Api mock

### PR DESCRIPTION
**Pull request by Waizik**

### CODE REVIEW

File: commands.ts [hunk 1]
```diff
@@ -1,49 +1,56 @@
...
+import { constants } from "./constants";
 
 Cypress.Commands.add(
     "apiMock",
-    (pattern: string, response: string | Object): Cypress.Chainable<string[]> => {
+    (
+        pattern: string,
+        response: string | Object,
+        serverAddressWithPort: string = `127.0.0.1:${constants.Port}`
+    ): Cypress.Chainable<void> => {
```
Comment (Waizik) [inline commands.ts]: Add timeout

Comment (Waizik) [reply commands.ts]: and logfalse

File: commands.ts [hunk 2]
```diff
@@ -1,49 +1,56 @@
...
         Cypress.log({ message: [pattern] });
         const options: IApiMockOptions = { pattern, response };
-        return (cy.task("api-mock:register", options, { log: false }) as any) as Cypress.Chainable<string[]>;
+        return (cy
+            .request("POST", `${serverAddressWithPort}${constants.Paths.registerMock}`, options)
```
Comment (Waizik) [inline commands.ts]: adress/constants

File: commands.ts [hunk 3]
```diff
@@ -1,49 +1,56 @@
...
-    (options: Partial<Cypress.Timeoutable> = {}): Cypress.Chainable<Map<string, IApiMockRequestData[]>> => {
-        Cypress.log({});
-        return cy.task("api-mock:get-requests", options, { log: false }).then((requests) => (requests as any) as Map<string, IApiMockRequestData[]>);
+    (
+        serverAddressWithPort: string = `127.0.0.1:${constants.Port}`,
+        options: Partial<Cypress.Timeoutable> = {}
+    ): Cypress.Chainable<Map<string, IApiMockRequestData[]>> => {
+        return cy
+            .request(`${serverAddressWithPort}${constants.Paths.getRequests}`, options)
+            .then((response) => (response.body as any) as Map<string, IApiMockRequestData[]>);
```
Comment (Waizik) [inline commands.ts]: Is it generic?

Comment (Waizik) [reply commands.ts]: Generickej typ primo na request

Comment (Waizik) [reply commands.ts]: A dale taky

File: commands.ts [hunk 4]
```diff
@@ -1,49 +1,87 @@
...
-        const options: IApiMockOptions = { pattern, response };
-        return (cy.task("api-mock:register", options, { log: false }) as any) as Cypress.Chainable<string[]>;
+        const data: IApiMockOptions = { pattern, response };
+        return (cy
+            .request({
+                log: false,
+                timeout: options?.timeout,
+                method: "POST",
+                url: `${serverAddressWithPort}${constants.Paths.registerMock}`,
+                body: data,
```
Comment (Waizik) [inline commands.ts:19]: options ...

File: constants.d.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+export declare const constants: {
+    Port: number;
+    Paths: {
```
Comment (Waizik) [inline constants.d.ts:2]: Delete

File: plugin/tsconfig.json [hunk 6]
```diff
@@ -3,5 +3,5 @@
     "compilerOptions": {
         "declaration": true
     },
-    "include": ["./**/*.ts"]
+    "include": ["./**/*.ts", "../constants.d.ts", "../constants.ts"]
```
Comment (Waizik) [inline plugin/tsconfig.json]: delete

File: readme.md [hunk 7]
```diff
@@ -38,6 +39,7 @@ beforeEach(() => cy.apiMockReset());
 ```
 
 You can also reset the log of requests and responses during the test suite.
+Requests, responses and mocks older then 5 minutes are deleted in 10 minutes interval by default.
```
Comment (Waizik) [inline readme.md]: Neplati, prepsat

File: server.d.ts [hunk 8]
```diff
@@ -0,0 +1,2 @@
+export declare function startAutoClearing(): void;
```
Comment (Waizik) [inline server.d.ts]: Delete

File: server.ts [hunk 9]
```diff
@@ -0,0 +1,275 @@
...
+let deleteTimeout: ReturnType<typeof setTimeout>;
+
+export function startServer(config: IApiMockConfiguration): void {
+    //Todo delete
+    setInterval(() => log(`Server on ${config.apiMockServer.hostname}:${config.apiMockServer.hostPort} is alive`, "\x1b[33m"), 150 * 1000);
+    const server = http.createServer(async (req, res) => {
+        log(`Server received url: ${req.url}`, "\x1b[33m");
+        try {
+            switch (req.url) {
+                // Todo delete
```
Comment (Waizik) [inline server.ts]: Klidne nechat cypress-api-mock-running

File: server.ts [hunk 10]
```diff
@@ -0,0 +1,275 @@
...
+    });
+}
+
+function setNewDeleteTimeout(): void {
+    try {
+        if (deleteTimeout !== undefined) {
+            clearTimeout(deleteTimeout);
+        }
+
+        deleteTimeout = setTimeout(() => reset(), 5 * 60000);
```
Comment (Waizik) [inline server.ts]: constants

File: server.ts [hunk 11]
```diff
@@ -0,0 +1,275 @@
...
+}
+
+function registerMock(pattern: string, response: string | Object): null {
+    log(`I\tMock registration for:\t${pattern} with response: ${response}`, "\x1b[33m");
+    mocks.set(pattern, response);
+    return null;
+}
+
+async function ProcessApiRequest(req: any, res: any): Promise<void> {
+    log("In process request");
```
Comment (Waizik) [inline server.ts]: Delete logs

File: server.ts [hunk 12]
```diff
@@ -0,0 +1,275 @@
...
+    resetMock();
+}
+
+function registerMock(pattern: string, response: string | Object): null {
+    log(`I\tMock registration for:\t${pattern} with response: ${response}`, "\x1b[33m");
+    mocks.set(pattern, response);
+    return null;
+}
+
+async function ProcessApiRequest(req: any, res: any): Promise<void> {
```
Comment (Waizik) [inline server.ts]: processAPi..

File: server.ts [hunk 13]
```diff
@@ -0,0 +1,275 @@
...
+    mocks.clear();
+}
+
+function reset(): void {
+    resetCalls();
+    resetMock();
+}
+
+function registerMock(pattern: string, response: string | Object): null {
+    log(`I\tMock registration for:\t${pattern} with response: ${response}`, "\x1b[33m");
```
Comment (Waizik) [inline server.ts]: Delete

File: server/startApiMockServer.ts [hunk 14]
```diff
@@ -0,0 +1,4 @@
+import { constants } from "../constants";
```
Comment (Waizik) [inline server/startApiMockServer.ts]: startServer.ts

File: server.ts [hunk 15]
```diff
@@ -0,0 +1,269 @@
...
+
+const mocks = new Map<string, string | Object>();
+const requests = new Map<string, IApiMockRequestData[]>();
+const responses = new Map<string, string[]>();
+
+let deleteTimeout: ReturnType<typeof setTimeout>;
+
+export function startServer(config: IApiMockConfiguration): void {
+    const server = http.createServer(async (req, res) => {
+        log(`Server received url: ${req.url}`, "\x1b[33m");
```
Comment (Waizik) [inline server.ts]: I\t

**Review by keeema — APPROVED** [review]

**Pull request merged**